nav2_tasks/include/nav2_tasks/task_client.hpp

```diff
@@ -57,26 +91,54 @@ class TaskClient
         std::bind(&TaskClient::onResultReceived, this, std::placeholders::_1));
     statusSub_ = node_->create_subscription<StatusMsg>(taskName + "_status",
         std::bind(&TaskClient::onStatusReceived, this, std::placeholders::_1));
+
+    return rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn::SUCCESS;
   }
 
-  TaskClient() = delete;
+  nav2_lifecycle::CallbackReturn on_activate(const rclcpp_lifecycle::State &) override
+  {
+    commandPub_->on_activate();
+    updatePub_->on_activate();
+    cancelPub_->on_activate();
 
-  ~TaskClient()
+    return rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn::SUCCESS;
+  }
+
+  nav2_lifecycle::CallbackReturn on_deactivate(const rclcpp_lifecycle::State &) override
   {
+    commandPub_->on_deactivate();
+    updatePub_->on_deactivate();
+    cancelPub_->on_deactivate();
+
+    return rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn::SUCCESS;
+  }
+
+  nav2_lifecycle::CallbackReturn on_cleanup(const rclcpp_lifecycle::State &) override
+  {
+    commandPub_.reset();
+    updatePub_.reset();
+    cancelPub_.reset();
+
+    resultSub_.reset();
+    statusSub_.reset();
+
+    return rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn::SUCCESS;
   }
 
   // The client can tell the TaskServer to execute its operation
   void sendCommand(const typename CommandMsg::SharedPtr msg)
   {
     resultReceived_ = false;
     statusReceived_ = false;
+
```

Comment:
Any reason you added a blank line here?

Reply:
It was just for formatting/readability. The file is now obsolete and deleted.